app/scripts/modules/core/src/pipeline/config/graph/pipeline.graph.component.ts

```diff
@@ -156,7 +156,10 @@ export class PipelineGraphController implements IController {
           (node: IPipelineNode) => 1 - sumBy(node.children, (child: IPipelineNode) => child.children.length),
           // great, same number of grandchildren, how about by nearest children, alphabetically by name, why not
           (node: IPipelineNode) => sortBy(node.children, 'phase').map((child: IPipelineNode) => [(child.phase - node.phase), child.name].join('-')).join(':'),
-          (node: IPipelineNode) => Number(node.id)
+          // if `id` is a number (or a string that maps to a number), sort above ids that are strings.
```

Comment:
comments this far into the sorting algorithm should convey a bit more exasperation